MsCorePkg/Library/PlatformBootManagerLib/BdsPlatform.c

```diff
@@ -388,6 +390,25 @@ PlatformBootManagerAfterConsole (
     case BOOT_ON_FLASH_UPDATE:
       EfiBootManagerConnectAll ();
       DEBUG ((DEBUG_INFO, "[%a] - signalling capsules are ready for processing\n", __FUNCTION__));
+      DEBUG ((DEBUG_INFO, "[%a] - Deleting Memory Type Information variable for capsule update\n", __FUNCTION__));
+      Status = gRT->GetVariable (

```

Comment:
I suggest setting `VariableSize` to `0` before calling `GetVariable()` so you're not dependent on random values picked up in stack memory as to whether `EFI_BUFFER_TOO_SMALL` will be returned.

Then, solely consider `EFI_BUFFER_TOO_SMALL` to indicate data is present and proceed with deletion.

Reply:
Updates.